src/containers/SagaMonitorView/SagaMonitorView.js

```diff
@@ -33,13 +36,43 @@ class SagaMonitorView extends React.Component {
   }
 
   updateFilter = () => this.setState(({ filter }) => (
-    { filter: { ...filter, word: this.filter.value } }
+    { filter: { word: this.filterWord.value, type: this.filterType.value } }
   ))
 
+  setFilterOptions = (items, { allCaps }) => {
+    const options = items.map(item => {
+      let label, value
+
+      if(typeof item === 'object') {
+        label = item.option || item.value
+        value = item.value
+      }
+      else if(typeof item === 'string') {
+        label = item
+        value = item
+      }
+
+      if (allCaps) label = label.toUpperCase()
+
+      return { label, value }
+
+    })
+  
+    this.setState({ filterOptions: options })
+  }
+
+  renderFilterOptions = () => this.state.filterOptions.map(option => {
+    return <FilterOption key={option.value} value={option.value}>{option.label}</FilterOption>
+  })
+
   renderCurrentView() {
     switch (this.state.currentView) {
       case EFFECT_VIEW:
-        return <EffectView rootEffectIds={this.props.rootEffectIds} filter={this.state.filter} />
+        return <EffectView
+          rootEffectIds={this.props.rootEffectIds}
+          filter={this.state.filter}
+          setFilterOptions={this.setFilterOptions}
+        />
```

Comment:
There is also another EffectView deep inside ActionView/Reactions/EffectView which shows a similar effect tree but only for those sagas that reacted to the selected action. Right now when selecting an action the app crashes because the deeply nested EffectView receive a null filter. So we should forward filter options from ActionView down to it

Reply:
Thanks for the review! I will fix it in the next PR.